### DISCUSSION TITLE
please help with configuration

### DISCUSSION THREAD

**Discussion by gregor3636** [RavenSystem/esp-homekit-devices#2194] [General]
how to change gpio4 to gpio5 in config
{"c":{"io":[[[0],6],[[4,5,12,13,14],7]],"pt":8,"w":10,"n ":"RGBW-CCT","b":[{"g":0,"t":5}]},"m":240,"a":[{"s":5,"t" :30,"g":[5],"b":[{"g":0}]}]}

**Comment by shadowgrapher** [comment]
As far as I can see there is only one mention of GPIO4 in second GPIO array and it's mode the same as GPIO5
```javascript
{
  "c":{
    "io":[
      [
        [
          0   /*** First array with only GPIO0 ***/
        ],
        6   /*** MODE 6 (Input only with binary input (button/switch) support) ***/
      ],
      [
        [
          4,   /*** Second array with GPIO4, GPIO5, GPIO12, GPIO13, GPIO14 ***/
          5,
          12,
          13,
          14
        ],
        7  /*** MODE 7 [Software PWM, only output] for all GPIOs in second array ***/
      ]
    ],
    "pt":8,
    "w":10,
    "n ":"RGBW-CCT",
    "b":[
      {
        "g":0,
        "t":5
      }
    ]
  },
  "m":240,
  "a":[
    {
      "s":5,
      "t":30,
      "g":[
        5
      ],
      "b":[
        {
          "g":0
        }
      ]
    }
  ]
}
```

**Reply by gregor3636** [reply]
Under GPIO0 there is a button connected under GPIO5, LED
and I would like to change it to gpio4

**Reply by shadowgrapher** [reply]
I assume GPIO4 is the white channel, GPIO5 is the little LED indicator. Do you want to make the white channel work as an indicator?

**Reply by gregor3636** [reply]
I want to replace gpio5 with gpio4
so that gpio4 takes over the functions of gpio5
and GPIO5 is not needed

**Reply by shadowgrapher** [reply]
```javascript
  "a":[
    {
      "s":5,
      "t":30,
      "g":[
        5         <---- GPIO5
       ]
``` 
In your accessories section change "g":[5] to "g":[4] so it will look like
```javascript
{"c":{"io":[[[0],6],[[4,5,12,13,14],7]],"pt":8,"w":10,"n ":"RGBW-CCT","b":[{"g":0,"t":5}]},"m":240,"a":[{"s":5,"t" :30,"g":[4],"b":[{"g":0}]}]}
```

**Reply by gregor3636** [reply]
Thank you very much

**Reply by gregor3636** [reply]
if it's not a problem, I have one more request to change the gpio??
replace gpio09 with gpio 03

{"c":{"io":[[[5,12,13],2],[[0,9,10],6]],"pt":8,"w":10,"n ":"Sonoff-Rolet","l":13,"b":[[10,5]]},"a":[{"t":45,"o":18,"c":14 ,"f":70,"0":{"r":[[12,1],[5]]},"1":{"r":[[12],[5,1]]} ,"2":{"r":[[12],[5]]},"3":{"r":[[12,0,0.2],[5]]},"4":{ "r":[[12],[5,0,0.2]]},"f3":[[0,0]],"f4":[[9,0]]}]}

is it enough to change this entry??
"f4":[[9,0]]}]}
"f 4":[[3,0]]}]}

<img width="232" alt="Zrzut ekranu 2024-01-20 o 12 01 07" src="https://github.com/RavenSystem/esp-homekit-devices/assets/133224745/1a96f4b9-46a8-4024-b730-a3d887b1f068">

**Reply by shadowgrapher** [reply]
Remember since v12, GPIO declarations must be in an array "io":[...]. 
You have two arrays in this section and one of them contains GPIO9.

> FYI: In your previous example we didn't touch that section because it already contains GPIO4 and GPIO5

```javascript
{
  "c":{
    "io":[
      [
        [
          5,
          12,
          13
        ],
        2
      ],
      [
        [
          0,
          9,         <---- GPIO9
          10
        ],
        6
      ]
    ],
```

**Reply by gregor3636** [reply]
so if I replace 9 with 3 here, it will be ok
{
  "c":{
    "io":[
      [
        [
          5,
          12,
          13
        ],
        2
      ],
      [
        [
          0,
          3,         <---- GPIO9 ??
          10
        ],
        6
      ]
    ],

**Reply by shadowgrapher** [reply]
Yes, if you need GPIO3 acts like a button:
```javascript
       [             
          0,         
          3,         
          10
        ],
        6            <----/ Mode 6 [Input only with binary input (button/switch) support]
      ]
```

**Reply by gregor3636** [reply]
{"c":{"io":[[[5,12,13],2],[[0,3,10],6]],"pt":8,"w":10,"n ":"Sonoff-Rolet","l":13,"b":[[10,5]]},"a":[{"t":45,"o":18,"c":14 ,"f":70,"0":{"r":[[12,1],[5]]},"1":{"r":[[12],[5,1]]} ,"2":{"r":[[12],[5]]},"3":{"r":[[12,0,0.2],[5]]},"4":{ "r":[[12],[5,0,0.2]]},"f3":[[0,0]],"f4":[[9,0]]}]}

ok??

**Reply by shadowgrapher** [reply]
...and state input `f4` I think
```javascript
{"c":{"io":[[[5,12,13],2],[[0,3,10],6]],"pt":8,"w":10,"n ":"Sonoff-Rolet","l":13,"b":[[10,5]]},"a":[{"t":45,"o":18,"c":14 ,"f":70,"0":{"r":[[12,1],[5]]},"1":{"r":[[12],[5,1]]} ,"2":{"r":[[12],[5]]},"3":{"r":[[12,0,0.2],[5]]},"4":{ "r":[[12],[5,0,0.2]]},"f3":[[0,0]],"f4":[[3,0]]}]}
```

**Reply by gregor3636** [reply]
Thank you very much for your help, everything works, best regards

**Reply by shadowgrapher** [reply]
Glad I was able to help :)
